src/probability/martingale/convergence.lean

```diff
@@ -231,4 +251,224 @@ mem_ℒp_limit_process_of_snorm_bdd
+
+Finally, to prove (c), we define $f_n := \mathbb{E}[h \mid \mathcal{F}_n]$. It is clear that
+$(f_n)_n$ is a martingale by the tower property for conditional expectations. Furthermore,
+$(f_n)_n$ is uniformly integrable in the probability sense. Indeed, as a single function is
+uniformly integrable in the measure theory sense, for all $\epsilon > 0$, there exists some
+$\delta > 0$ such that for all measurable set $A$ with $\mu(A) < δ$, we have
+$\mathbb{E}|h|\mathbf{1}_A < \epsilon$. So, since for sufficently large $\lambda$, by the Markov
+inequality, we have for all $n$,
+$$
+  \mu(|f_n| \ge \lambda) \le \lambda^{-1}\mathbb{E}|f_n| \le \lambda^{-1}\mathbb|g| < \delta,
+$$
+we have for sufficently large $\lambda$, for all $n$,
+$$
+  \mathbb{E}|f_n|\mathbf{1}_{|f_n| \ge \lambda} \le
+    \mathbb|g|\mathbf{1}_{|f_n| \ge \lambda} < \epsilon,
+$$
+implying $(f_n)_n$ is uniformly integrable. Now, to prove $f_n \to h$ almost everywhere and in
+L¹, it suffices to show that $h = g$ almost everywhere where $g$ is the almost everywhere and L¹
+limit of $(f_n)_n$ from part (b) of the theorem. By noting that, for all $s \in \mathcal{F}_n$,
+we have
+$$
+  \mathbb{E}g\mathbf{1}_s = \mathbb{E}[\mathbb{E}[g \mid \mathcal{F}_n]\mathbf{1}_s] =
+    \mathbb{E}[\mathbb{E}[h \mid \mathcal{F}_n]\mathbf{1}_s] = \mathbb{E}h\mathbf{1}_s
+$$
+where $\mathbb{E}[g \mid \mathcal{F}_n = \mathbb{E}[h \mid \mathcal{F}_n]$ almost everywhere
+by part (b); the equality also holds for all $s \in \mathcal{F}_\infty$ by Dynkin's theorem.
+Thus, as both $h$ and $g$ are $\mathcal{F}_\infty$-measurable, $h = g$ almost everywhere as
+required.
+
+Similar to the a.e. martingale convergence theorem, rather than showing the existence of the
+limiting process, we phrased the L¹-martingale convergence theorem by proving that a submartingale
+does converge in L¹ to its `limit_process`. However, in contrast to the a.e. martingale convergence
+theorem, we do not need to introduce a L¹ version of `filtration.limit_process` as the L¹ limit
+and the a.e. limit of a submartingale coincide.
+
+-/
+
+/-- Part a of the **L¹ martingale convergence theorem**: a uniformly integrable submartingale
+adapted to the filtration `ℱ` converges a.e. and in L¹ to an integrable function which is
+measurable with respect to the σ-algebra `⨆ n, ℱ n`. -/
+lemma submartingale.tendsto_snorm_one_limit_process
+  (hf : submartingale f ℱ μ) (hunif : uniform_integrable f 1 μ) :
+  tendsto (λ n, snorm (f n - ℱ.limit_process f μ) 1 μ) at_top (𝓝 0) :=
+begin
+  obtain ⟨R, hR⟩ := hunif.2.2,
+  have hmeas : ∀ n, ae_strongly_measurable (f n) μ :=
+    λ n, ((hf.strongly_measurable n).mono (ℱ.le _)).ae_strongly_measurable,
+  exact tendsto_Lp_of_tendsto_in_measure _ le_rfl ennreal.one_ne_top hmeas
+    (mem_ℒp_limit_process_of_snorm_bdd hmeas hR) hunif.2.1
+    (tendsto_in_measure_of_tendsto_ae hmeas $ hf.ae_tendsto_limit_process hR),
+end
+
+lemma submartingale.ae_tendsto_limit_process_of_uniform_integrable
```

Comment:
```suggestion
lemma submartingale.ae_tendsto_limit_process
```
That's shorter, and similar to `submartingale.tendsto_snorm_one_limit_process`, which does not mention the `uniform_integrable` hypothesis.

Reply:
We already have 
```lean
lemma submartingale.ae_tendsto_limit_process [is_finite_measure μ]
  (hf : submartingale f ℱ μ) (hbdd : ∀ n, snorm (f n) 1 μ ≤ R) :
  ∀ᵐ ω ∂μ, tendsto (λ n, f n ω) at_top (𝓝 (ℱ.limit_process f μ ω))
```
which is why I added the `uniform_integrable` into the name

Reply:
Let's keep it then.